api_volontaria/apps/position/test_view_positions.py

```diff
@@ -0,0 +1,242 @@
+        self.check_attributes(content)
+
+    def test_update_position_as_admin(self):
+        """
+        Ensure we can update a position if we are an admin.
+        """
+        new_hourly_wage = 16
+        data_post = {
+            'hourly_wage': new_hourly_wage,
+        }
+
+        self.client.force_authenticate(user=self.admin)
+
+        response = self.client.patch(
+            reverse(
+                'position-detail',
+                kwargs={
+                    'pk': self.position.id
+                },
+            ),
+            data_post,
+            format='json',
+        )
+
+        content = json.loads(response.content)
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.check_attributes(content)
+        self.assertEqual(
+            content['hourly_wage'],
+            new_hourly_wage,
+        )
+
+    def test_update_position(self):
+        """
+        Ensure we can't update a position if we are a simple user.
+        """
+        new_hourly_wage = 16
+        data_post = {
+            'hourly_wage': new_hourly_wage,
+        }
+
+        self.client.force_authenticate(user=self.user)
+
+        response = self.client.patch(
+            reverse(
+                'position-detail',
+                kwargs={
+                    'pk': self.position.id
+                },
+            ),
+            data_post,
+            format='json',
+        )
+
+        content = json.loads(response.content)
+
+        self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
+        self.assertEqual(
+            content,
+            {
+                'detail': 'You do not have permission to perform this action.'
+            }
+        )
+
+    def test_delete_position_as_admin(self):
+        """
+        Ensure we can delete a position if we are an admin.
+        """
+        self.client.force_authenticate(user=self.admin)
+
+        response = self.client.delete(
+            reverse(
+                'position-detail',
+                kwargs={
+                    'pk': self.position.id
+                },
+            )
+        )
+
+        self.assertEqual(response.status_code, status.HTTP_204_NO_CONTENT)
+        self.assertEqual(response.content, b'')
+
+    def test_position_event(self):
+        """
+        Ensure we can't delete a position if we are a simple user.
+        """
+        self.client.force_authenticate(user=self.user)
+
+        response = self.client.patch(
+            reverse(
+                'position-detail',
+                kwargs={
+                    'pk': self.position.id
+                },
+            )
+        )
+
+        content = json.loads(response.content)
+
+        self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
+        self.assertEqual(
+            content,
+            {
+                'detail': 'You do not have permission to perform this action.'
+            }
+        )
```

Comment:
- Name of the test is not accurate
 - I think that it should be a delete as a user (i guess from the logic of repetition in others tests)
  - If it's the case, it should be `self.client.delete` and not `self.client.patch`

Reply:
- test renamed to "test_delete_position"
- agreed: it is a delete as a user
- edited to self.client.delete